### ISSUE TITLE
Adding annotation text would disappear after searching

### ISSUE THREAD

**Issue by Moyf** [alberti42/obsidian-plugins-annotations#5]
And I found another problem that if I search the Plugin, the annotation placeholder would be lost.
![demo2](https://github.com/alberti42/obsidian-plugins-annotations/assets/4013062/dbc32ddd-cadd-491b-af32-b17f76a3189c)

_Originally posted by @Moyf in https://github.com/alberti42/obsidian-plugins-annotations/issues/4#issuecomment-2216114738_

**Comment by alberti42** [comment]
Thanks! 

I still see the double lock. Is the problem really solved with the double lock?

Actually, what you are now reporting was the problem I solved with the release 1.4.0 of the plugin. Did you try to restart Obsidian and see whether this behavior is still there?

**Comment by Moyf** [comment]
> Thanks!
> 
> I still see the double lock. Is the problem really solved with the double lock?
> 
> Actually, what you are now reporting was the problem I solved with the release 1.4.0 of the plugin. Did you try to restart Obsidian and see whether this behavior is still there?

Oh, the double locks is because this image is recorded yesterday - this new issue just reference to the original post, as well as the image.

![image](https://github.com/alberti42/obsidian-plugins-annotations/assets/4013062/d5d61c40-f1a1-4ef7-b589-56957740b2f1)
I've restarted my OB several times after this version's plugin (2 days ago) 🤔

**Comment by Moyf** [comment]
![demo](https://github.com/user-attachments/assets/4d4d39b9-1325-4840-8b05-17269024e3e2)

Just re-install the plugin and the problem seems still exists 😭

---

![demo](https://github.com/user-attachments/assets/9d7278c1-5acd-48e4-bdd8-a60e687a2da3)

Also tested in Sandbox vault, got the same issue

**Comment by ksdavidc** [comment]
Was just about to post the same thing because I had noticed it, but somehow now it is partially working for me.
Partially? I can see existing annotations, but can't create new ones.

Could have been related to another weird behavior, namely I was getting 2 "Lock" icons, so I will post more details elsewhere.

**Comment by ksdavidc** [comment]
I hate to say it, but my annotations are disappearing again after search (and only after search) and then they are completely inaccessible (i.e. clicking on them leads to the settings page, not allows annotation). However it is only in one of my vaults, and I have too many plugins to dissect which one might be in conflict with it.

https://github.com/user-attachments/assets/9acec212-4ea1-4df9-aaab-7e01eaff836f

I tried resetting the plugin settings to their defaults, but still not working. Search kills it.

```
SYSTEM INFO:
	Obsidian version: v1.6.7
	Installer version: v1.6.5
	Operating system: Darwin Kernel Version 23.6.0: Mon Jul 29 21:13:00 PDT 2024; root:xnu-10063.141.2~1/RELEASE_X86_64 23.6.0
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Base theme: dark
	Community theme: AnuPpuccin v1.5.0
	Snippets enabled: 21
	Restricted mode: off
	Plugins installed: 233
	Plugins enabled: 182
		1: Callout Suggestions v1.0.2
		2: Admonition v10.3.2
		3: Advanced Tables v0.22.0
		4: Another Quick Switcher v11.6.0
		5: Apply Patterns v2.1.2
		6: Auto Split v1.2
		7: Automatic Table Of Contents v1.4.0
		8: Better Command Palette v0.17.1
		9: Better Export PDF v1.9.2
		10: Calendar v1.5.10
		11: Better footnote v1.0.1
		12: Better PDF 2 v2.0.3
		13: Better Search Views v0.2.2
		14: Better Word Count v0.10.1
		15: Blockier v1.1.0
		16: BRAT v1.0.3
		17: Canvas Card Background Remover v0.1.4
		18: CardBoard v0.7.9
		19: Change Case v2.0.106
		20: Colorful Tag v1.3.2
		21: Doubleshift v2.2.1
		22: Cloze v0.1.17
		23: Colored Tags Wrangler v0.19.3
		24: Commander v0.5.1
		25: Convert url to preview (iframe) v0.5.0
		26: Copy as HTML v1.1.3
		27: Copy document as HTML v0.8.1
		28: Crumbs v0.1.2
		29: CSV Table v1.2.0
		30: Custom Classes v2.6.1
		31: Custom Frames v2.4.7
		32: Dangling links v0.3.0
		33: DataLoom v8.16.1
		34: Dataview v0.5.67
		35: Dynbedded v1.0.0
		36: Easy toggle sidebars v2.0.1
		37: Badges v1.1.0
		38: Create Note in Folder v1.12.2
		39: Advanced URI v1.44.1
		40: Copilot v2.6.11
		41: Advanced Paste v2.7.0
		42: Booksidian v0.5.2
		43: Control Characters v0.3.0
		44: Auto Embed v1.2.2
		45: Zoom v1.1.2
		46: Yesterday v1.2.0
		47: Wrap with shortcuts v1.3.0
		48: Wordy v1.3.1
		49: VSCode Editor v1.0.4
		50: Editing Toolbar v2.4.16
		51: Emoji Toolbar v0.4.1
		52: Enhanced tables v1.0.5
		53: Export Image plugin v2.3.13
		54: Extract PDF Annotations v1.4.0
		55: Extract url content v0.12.1
		56: File Diff v1.1.1
		57: File Explorer Note Count v1.2.2
		58: File Info Panel v1.3
		59: File Tree Alternative v2.6.0
		60: Find & Replace in Selection v1.1.0
		61: Find orphaned files and broken links v1.10.1
		62: Focus and Highlight v1.3.2
		63: Folder Note Core v1.3.5
		64: Footlinks v0.1.7
		65: Footnote & Citation Indicator v0.3.5
		66: Footnote Shortcut v0.1.3
		67: Garble Text v1.2.0
		68: Get Info v1.1.0
		69: HelpMate v1.0.6
		70: Hide Sidebars on Window Resize v1.1.0
		71: Hider v1.5.1
		72: Image Converter v1.1.7
		73: Hover Editor v0.11.20
		74: Hover External Link Plugin v1.0.0
		75: Hyphenation v0.1.0
		76: Icon Shortcodes v0.9.7
		77: Image in Editor v2.2.6
		78: Interlinear Glossing v1.5.0
		79: Key-Value List v1.7.0
		80: LanguageTool Integration v0.3.7
		81: Lapel v0.1.5
		82: Lineage v0.6.1
		83: Link Tree v1.0.0
		84: Links v1.17.21
		85: Linter v1.26.0
		86: Local File Interface v0.1.0
		87: Local Images Plus v0.15.9
		88: Local REST API v2.5.4
		89: MAKE.md v0.9.1007
		90: Merge Notes v0.0.9
		91: Markdown Table Editor v0.3.1
		92: Markdown Formatting Assistant v0.4.1
		93: Style Settings v1.0.9
		94: Open Link With v0.1.10
		95: Open Plugin Settings v1.3.2
		96: Open with v0.3.2
		97: Outliner v4.8.1
		98: OZ Calendar v0.3.4
		99: Vault Explorer v1.46.0
		100: UNITADE.md v2.4.0
		101: Unicode Search v0.6.1
		102: Translate v1.4.9
		103: Time Ruler v2.5.1
		104: Toggle Case v1.1.0
		105: Tidy Footnotes v0.1.1
		106: Thino v2.4.57
		107: Theme Toggler v1.0.1
		108: Theme Picker v1.0.8
		109: Text Transporter v1.0.11
		110: Text Snippets v0.1.2
		111: Text Format v3.1.0
		112: Text Extractor v0.5.2
		113: Text expand v0.11.4
		114: Templater v2.8.0
		115: Tasks Calendar Wrapper v0.3.1
		116: Tasks v7.11.0
		117: Task Collector (TC) v1.1.4
		118: TagFolder v0.18.7
		119: Tag Wrangler v0.6.1
		120: Tag Project v0.9.0
		121: Tag & Word Cloud v1.4.1
		122: Table of Contents v0.2.0
		123: Table Generator v1.4.1
		124: Paste Mode v5.0.2
		125: Paste URL into selection v1.7.0
		126: PDF break page v1.0.0
		127: Periodic Notes v0.0.17
		128: Studier v1.0.1
		129: Strange New Worlds v2.1.4
		130: Sort & Permute lines v0.7.0
		131: Snippetor v0.4.5
		132: Slides Extended v2.1.3
		133: Slide Note v0.0.16
		134: Size History v1.0.0
		135: Show Current File Path v0.5.2
		136: Sheets Extended v1.2.10
		137: Share Note v0.8.17
		138: Share my plugin list v0.3.3
		139: Scratchblocks v0.0.1
		140: Reveal Active File Button v2.0.3
		141: Reset Font Size v1.0.2
		142: Related Notes Finder v1.1.0
		143: Regex Find/Replace v1.2.0
		144: Refresh Preview v1.3.0
		145: Quiet Outline v0.3.36
		146: QuickAdd v1.11.1
		147: Quick Tagger v2.2.3
		148: Quick Switcher++ v4.5.1
		149: Quick Explorer v0.2.10
		150: Query Control v0.6.0
		151: Projects v1.17.4
		152: PopKit v1.0.13
		153: Plugin Update Tracker v1.5.2
		154: Plugin Manager v0.1.5
		155: Plugin Groups v2.1.0
		156: Pluck v1.0.1
		157: Plaintext v0.3.0
		158: Personal Assistant v1.4.1
		159: Outliner.md v0.1.9
		160: floating toc v2.4.7
		161: Enhanced Annotations v0.1.6
		162: Hotkey Helper v0.3.18
		163: Kanban v2.0.51
		164: Excalidraw v2.5.1
		165: Better Comment Toggle v1.0.3
		166: Ink v0.2.6
		167: Outline Converter v1.3.1
		168: Digital Garden v2.57.2
		169: Datepicker v0.3.23
		170: Natural Language Dates v0.6.2
		171: PDF++ v0.40.11
		172: MySnippets v1.2.3
		173: Note Refactor v1.8.2
		174: Omnisearch v1.24.1
		175: Card Navigator v1.0.8
		176: Canvas Explorer v1.0.6
		177: Edit mode switch v0.0.3
		178: Enhance YouTube Links v1.0.8
		179: Tagged Documents Viewer v1.0.2
		180: Tag Summary v2.1.5
		181: Plugins Annotations v1.7.2
		182: Settings Management v1.0.5

```

**Comment by alberti42** [comment]
Hey [ksdavidc](https://github.com/ksdavidc), I am sorry that you experience this trouble. It is hard to say what the culprit is. You have 182 plugins installed ;-) Perhaps the culprit is some other plugin? Or perhaps my plugin where there are some edge cases that are not covered properly.

I could give it a try to look into the problem but you have to share your vault. You would have to make a copy where you remove all your sensitive information. I believe, in the backup copy you can delete all your notes. I don't believe those are the problem. Then you should also check that no other sensitive data is stored in the configuration folder `.obsidian` inside your vault. Sometimes there are plugins requiring your username and password to access a particular service through API. You should remove this kind of stuff.

Finally you should test if the copy of the original vault still displays the problem. If so, it would be great if you make a zip file and send it over. Feel free to share it with me personally to minimize the chance of any leak of private information.

You would understand, without a duplicate of your vault from you, I will not install myself 182 plugins to reproduce your bug ;-)

Cheers, Andrea

**Comment by ksdavidc** [comment]
Yes, I quite understand I haven't given you enough information. This was mostly just a heads-up that something is still buggy somewhere, though you are perfectly right that it likely implicates another plugin. (the idea being you or others would have a chance to observe or share clues you might uncover or think of in the course of your other efforts. For example, the only thing I feel fairly certain about is that it involves search, which made me wonder if you would have any idea where that would intersect with your plugin.) Despite the number of plugins I have, when I run I usually get low integer errors, if any. One of the features of this bug is that it throws no errors, as bugs usually do, eventually. That could bode well for the idea it is not your plugin at all. 

I would never expect you to install 182 plugins. 180 maybe ;-)

I am in the process of winnowing out excess plugins (which process of course your plugin was part of). (Ironically, this makes other bugs appear. Many plugins are dependent on each other). Also, I have yet to install the upcoming update, so that might fix it. 

The reason I have so many is that this was actually my first vault, which became, due to its nature, a multipurpose vault. I was young and inexperienced and accumulated a lot of experimental stuff along the way.

If I still detect a bug after the purge, I will send you a secure minimal working example. If I do find the culprit, I'll let you know.

You've been very kind, and I really appreciate the effort you put in on my behalf, and the plugin itself. Thank you.

**Comment by alberti42** [comment]
Dear David,

If at some point you can narrow it down to a few plugins, that would be great. I will be happy to take a closer look. I don't think you need to worry too much about the rest, because this feature only seems to affect the visualisation when you do a search, and only in your vault with many plugins. Granted. It would be nice to know why. Maybe there is another plugin that misbehaves, maybe I can improve my plugin. It is always good to remove bugs, even the small ones. They might become big in a future update.

I will close the issue for now, but please reopen it if you find more clues or can narrow down the selection.

**Comment by ksdavidc** [comment]
Believe it or not, I found it. Hotkey helper  prevents annotations on search. MWE attached.
[test annotations.zip](https://github.com/user-attachments/files/17436774/test.annotations.zip)
I did get one, unofortunately transient msg about not being able to get id in my sandbox vault, but it did not happen when I created the vault from scratch

**Comment by alberti42** [comment]
Hey David, thank you! I imagine it took you some time to find out what the origin of the problem was. I confirm I can reproduce the bug. 

Can you explain to me what the Hotkey plugin does? I tried to install it, but the plugin has no effect on my vault. I disabled all other plugins and yet no effect.

It claims it shows a button for quickly accessing the menu to configure the shortcuts of the plugins. But this is something provided by the plain vanilla version of Obsidian. Why would you need a plugin to show these buttons? Here is an example what I see in my vault without installing the Hotkey plugin:

![Bildschirmfoto 2024-10-18 um 21 30 42](https://github.com/user-attachments/assets/ea67457a-e4c7-4d1f-acdd-281adc06d13f)

Could it be that the plugin Hotkey is buggy? Before I dive in searching for the broken code, I want to understand what the plugin is supposed to do.

**Comment by alberti42** [comment]
Ok. The warning you found on the console was not specific to the Hotkey plugin, but concerned any plugin when they are installed for the first time. I fixed the mini bug and the warning is no longer there. My plugin is now aware when new plugins are installed and updates the list of installed plugins, which is maintained internally to my plugin.

I also fixed some other minor bugs. Hopefully, I did not open new ones. I will post a new release, likely during the weekend.

I think I can run the Hotkeys plugin. For me, the only thing it does, it shows how many hotkeys are configured. Maybe I'm missing the purpose of this plugin. I will try to fix the conflict.

**Comment by alberti42** [comment]
Hi David,

I decided to release a new version with small bug fixes. I suggest we close this issue. I very strongly believe that the problem is in the code of "Hotkeys" plugin. Hopefully, I will not get criticized for that, but it does not look to be programmed in a clean way. It seems full of fixes and patches. I could see in the code of that plugin that functions were overwritten. If this is not done properly, it causes the problem you are seeing. I am speaking about functions concerning the visualization of the plugins in the community preference pane when you make a search.

I saw you filed a request for a bug fix at the Hotkey plugin github page. I hope they will come up with a solution, as I believe the problem resides there.

I close the issue. Feel free to reopen it.

**Comment by ksdavidc** [comment]
Please do close it. I am sorry to have made you go to so much trouble. I have little doubt you are right about hotkey, especially given what you say about how the code is written. I'll defend you in court!

The plugin doesn't create the buttons (though that is what it says in its readme). One thing it does, that was slightly useful, is that it adds color coding and tooltips that indicate conflicts and the number of keys used. 
<img width="293" alt="Screenshot 2024-10-19 at 6 56 15 AM" src="https://github.com/user-attachments/assets/26284dba-6b26-43eb-9fb6-7ad7b60e6a2b">

Some of its functions, like the plugin URL have been superseded by obsidian itself. 

Installed your update and Hotkey is still blocking on search, so I've disabled it for now. 

It is just really interesting for me that with my hundreds of plugins, this is the only one that got zapped by hotkey, AFAICT, and why??? Time will tell, I guess.

**Comment by alberti42** [comment]
Ok. I did some more debugging of the Hotkey plugin. I can confirm that it is not written in a proper manner. Please talk to the developer. The problem is in the following function. I paste all of it, but you have only to appreciate a couple of points I will comment about later

```typescript
 // Add top-level items (search and pseudo-plugins)
    addGlobals(tabId: string, settingEl: HTMLDivElement) {
        this.globalsAdded = true;

        // Add a search filter to shrink plugin list
        const containerEl = settingEl.parentElement;
        let searchEl: SearchComponent;
        if (tabId !== "plugins" || this.searchInput) {
            // Replace the built-in search handler
            (searchEl = this.searchInput)?.onChange(changeHandler);
        } else {
            const tmp = new Setting(containerEl).addSearch(s => {
                searchEl = s;
                s.setPlaceholder("Filter plugins...").onChange(changeHandler);
            });
            searchEl.containerEl.style.margin = "0";
            containerEl.createDiv("hotkey-search-container").append(searchEl.containerEl);
            tmp.settingEl.detach();
        }
        if (tabId === "community-plugins") {
            searchEl.inputEl.addEventListener("keydown", e => {
                if (e.keyCode === 13 && !Keymap.getModifiers(e)) {
                    this.gotoPlugin();
                    return false;
                }
            })
        }
        const plugin = this;
        function changeHandler(seek: string){
            const find = (plugin.lastSearch[tabId] = seek).toLowerCase();
            function matchAndHighlight(el: HTMLElement) {
                if (!el) return false;
                const text = el.textContent = el.textContent; // clear previous highlighting, if any
                const index = text.toLowerCase().indexOf(find);
                if (!~index) return false;
                el.textContent = text.substr(0, index);
                el.createSpan("suggestion-highlight").textContent = text.substr(index, find.length);
                el.insertAdjacentText("beforeend", text.substr(index+find.length))
                return true;
            }
            containerEl.findAll(".setting-item").forEach(e => {
                const nameMatches = matchAndHighlight(e.find(".setting-item-name"));
                const descMatches = matchAndHighlight(
                    e.find(".setting-item-description > div:last-child") ??
                    e.find(".setting-item-description")
                );
                const authorMatches = matchAndHighlight(
                    e.find(".setting-item-description > div:nth-child(2)")
                );
                e.toggle(nameMatches || descMatches || authorMatches);
            });
        }
        defer(() => {
            if (!searchEl) return
            if (searchEl && typeof plugin.lastSearch[tabId] === "string") {
                searchEl.setValue(plugin.lastSearch[tabId]);
                searchEl.onChanged();
            }
            if (!Platform.isMobile) searchEl.inputEl.select();
        });
        containerEl.append(settingEl);

        if (tabId === "plugins") {
            const editorName    = this.getSettingsTab("editor")?.name || "Editor";
            const workspaceName = this.getSettingsTab("file")?.name   || "Files & Links";
            this.createExtraButtons(
                new Setting(settingEl.parentElement)
                    .setName("App").setDesc("Miscellaneous application commands (always enabled)"),
                {id: "app", name: "App"}, true
            );
            this.createExtraButtons(
                new Setting(settingEl.parentElement)
                    .setName(editorName).setDesc("Core editing commands (always enabled)"),
                {id: "editor", name: editorName}, true
            );
            this.createExtraButtons(
                new Setting(settingEl.parentElement)
                    .setName(workspaceName).setDesc("Core file and pane management commands (always enabled)"),
                {id: "workspace", name: workspaceName}, true
            );
            settingEl.parentElement.append(settingEl);
        }
    }
```

The problem is with the lines of code:
```
// Replace the built-in search handler
            (searchEl = this.searchInput)?.onChange(changeHandler);
```

It totally replaces the original Obisidian handler for the visualization of the search results with a new function: 
```
function changeHandler(seek: string){
...
...
```

This way of coding is dangerous because it has hard-coded a pseudo duplicate of the original Obsidian code, with some patches. It is not the right way to do it.

**Comment by ksdavidc** [comment]
wow, that's great. I'll pass it on and let the other developer take it from there. For me it is not critical, but knowing where the source of the problem is makes it more likely it can be fixed. Thank you so much for all your insight and effort.

**Comment by pjeby** [comment]
I've fixed the interaction in Hotkey Helper 0.3.19; thanks for giving a heads-up to the specific location involved, as I hadn't gotten around to that yet.

Please note that HH is an older plugin that was written *before* Obsidian had the nice little buttons -- that was a feature directly inspired by Hotkey Helper.  But they didn't copy the whole thing, as Obsidian *doesn't* show hotkey counts and conflicts the way HH does!

The clash over search is precisely because of this feature: when Obsidian added in its own nice little buttons, they didn't include hotkey conflict management, and they changed the search to re-render all the plugins, which broke HH's coloring and tooltips.  So I had to put back Obsidian's older way of plugin searching in order to put back HH's hotkey info in a timely fashion.

I won't go into the dozen or so minor features and bug fixes HH applies to Obsidian, as that's not really relevant here.  I just want to note that I literally asked the Obsidian devs if they would just address those issues wholesale shortly after writing HH (so I could do away with HH entirely), and they chose to superficially imitate *some* of them, one at a time, over many releases, mostly without mention in release notes, and usually in such a way that HH can't just *stop* implementing the feature, but instead has to jump through bigger hoops to implement the *parts* of its features they left out in their implementations.   I would much prefer to do things differently, but this is where we are.

**Comment by alberti42** [comment]
Hi PJ,

Thanks a lot for telling us this interesting story. Yes, I share your feeling that Obsidian team can sometimes be frustrating.

I really do not understand why they do not release most of the code open source and work more closely with the huge community they have. They are wasting so many opportunities where developers work essentially for free for them. They could focus on a few products to make revenues, such as publishing Obisidian online and syncing Obsidian notes. I am regularly paying my subscription. And I would do even more so if the code was open sourced.

Thanks @ksdavidc for the valuable debugging feedback and persistence in killing big and small bugs.
